tests/test_target.py

```diff
@@ -37,23 +36,57 @@ def build_targets():
 
     targets = []
 
+    if target_configs.get('AndroidTarget') is not None:
+        print('> Android targets:')
+        for entry in target_configs['AndroidTarget'].values():
+            pp(entry)
+            a_target = AndroidTarget(
+                connection_settings=entry['connection_settings'],
+                conn_cls=lambda **kwargs: AdbConnection(adb_as_root=True, **kwargs),
+            )
+            targets.append((a_target, None))
+
+    if target_configs.get('LinuxTarget') is not None:
+        print('> Linux targets:')
+        for entry in target_configs['LinuxTarget'].values():
+            pp(entry)
+            l_target = LinuxTarget(connection_settings=entry['connection_settings'])
+            targets.append((l_target, None))
+
     if target_configs.get('LocalLinuxTarget') is not None:
         print('> LocalLinux targets:')
         for entry in target_configs['LocalLinuxTarget'].values():
             pp(entry)
             ll_target = LocalLinuxTarget(connection_settings=entry['connection_settings'])
-            targets.append(ll_target)
+            targets.append((ll_target, None))
+
+    if target_configs.get('QEMUTargetRunner') is not None:
+        print('> QEMU target runners:')
+        for entry in target_configs['QEMUTargetRunner'].values():
+            pp(entry)
+            qemu_settings = entry.get('qemu_settings') and entry['qemu_settings']
+            connection_settings = entry.get(
+                'connection_settings') and entry['connection_settings']
+
+            qemu_runner = QEMUTargetRunner(
+                qemu_settings=qemu_settings,
+                connection_settings=connection_settings,
+            )
+            targets.append((qemu_runner.target, qemu_runner))
```

Comment:
is there a point in returning tuples rather than just the runner, considered you can get the target by using `runner.target` ?

Reply:
You cannot if we want to test ChromeOS with QEMU, i.e., `targets.append((c_target, qemu_runner))`.

Reply:
As commented below/offline, this should be achievable with an implementation of a runner that allows sharing a target with another runner and only disposing of the Target once both are finished with it.